app/controllers/blogs_controller.rb

```diff
@@ -2,14 +2,15 @@
 
 class BlogsController < ApplicationController
   skip_before_action :authenticate_user!, only: %i[index show]
-
-  before_action :set_blog, only: %i[show edit update destroy]
+  before_action :set_blog, only: %i[edit update destroy]
 
   def index
     @blogs = Blog.search(params[:term]).published.default_order
   end
 
-  def show; end
+  def show
+    @blog = Blog.where(user_id: current_user&.id).or(Blog.published).find(params[:id])
```

Comment:
belongs_to をベースに検索する場合は、オブジェクトを指定できると思いますー。

```suggestion
    @blog = Blog.where(user: current_user).or(Blog.published).find(params[:id])
```